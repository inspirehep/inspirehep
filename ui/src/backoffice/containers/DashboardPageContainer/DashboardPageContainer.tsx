import React, { useEffect, useCallback, useMemo } from 'react';
import classNames from 'classnames';
import { Card, Input, Select, Tabs } from 'antd';
import { Link } from 'react-router-dom';
import { Action, ActionCreator } from 'redux';
import { connect, RootStateOrAny } from 'react-redux';
import { List, Map } from 'immutable';

import './DashboardPageContainer.less';
import Breadcrumbs from '../../components/Breadcrumbs/Breadcrumbs';
import { BACKOFFICE_SEARCH } from '../../../common/routes';
import {
  isUserLoggedInToBackoffice,
} from '../../../actions/backoffice';
import EmptyOrChildren from '../../../common/components/EmptyOrChildren';
import LoadingOrChildren from '../../../common/components/LoadingOrChildren';
import { COLLECTIONS, getIcon, handleSearch } from '../../utils/utils';
import { searchQueryUpdate } from '../../../actions/search';
import { BACKOFFICE_SEARCH_NS } from '../../../search/constants';

interface DashboardPageContainerProps {
  dispatch: ActionCreator<Action>;
  facets: Map<string, any>;
  loading: boolean;
  query: Map<string, any>;
}

const { Option } = Select;
const { Search } = Input;

const TEXT_CENTER: Record<string | number, string & {}> = {
  textAlign: 'center',
};

const DashboardPageContainer: React.FC<DashboardPageContainerProps> = ({
  dispatch,
  query,
  facets,
  loading,
}) => {
  useEffect(() => {
    dispatch(isUserLoggedInToBackoffice());
  }, [dispatch]);

  const workflowTypes: List<Map<string, any>> = useMemo(() => {
    return facets?.getIn([
      '_filter_workflow_type',
      'workflow_type',
      'buckets',
    ]) as List<Map<string, any>>;
  }, [facets]);



  const renderWorkflowStatus = useCallback(
    (type: Map<string, any>) =>
      (type?.getIn(['status', 'buckets']) as List<any>)?.map((status) => (
        <a
          href={`${BACKOFFICE_SEARCH}?workflow_type=${type?.get('key')}&status=${status?.get('key')}`}
          key={status?.get('key')}


        >
          <div
            className={classNames(
              'flex justify-between',
              status?.get('key')?.toLowerCase()
            )}
          >
            <p className="ttc">
              {getIcon(status?.get('key'))}
              {status?.get('key')}
            </p>
            <span className="b">{status?.get('doc_count')}</span>
          </div>
        </a>
      )),
    [dispatch]
  );

  const renderWorkflowCards = useMemo(
    () =>
      workflowTypes?.map((type: Map<string, any>) => (
        <Card
          title={
            <div>
              <p>
                {
                  COLLECTIONS.find(
                    (collection) => collection?.value === type?.get('key')
                  )?.key
                }
              </p>
              <p className={classNames('f2 mb0 black')}>
                {type?.get('doc_count')}
              </p>
              <Link
                to={`${BACKOFFICE_SEARCH}?workflow_type=${type?.get('key')}`}
                className="normal f6"

              >
                View all
              </Link>
            </div>
          }
          headStyle={TEXT_CENTER}
          className={COLLECTIONS.find(
            (collection) => collection?.value === type?.get('key')
          )
            ?.key.toLowerCase()
            .replace(/ /g, '-')}
          key={type?.get('key')}
        >
          {renderWorkflowStatus(type)}
        </Card>
      )),
    [workflowTypes, renderWorkflowStatus, dispatch]
  );

  const tabItems = useMemo(
    () => [
      {
        label: <h3>Collections</h3>,
        key: '1',
        children: (
          <>
            <Link
              to={BACKOFFICE_SEARCH}
              className="db w-100 tc f5 mt4"
              data-testid="view-all"
            >
              View all
            </Link>
            <div className="cards-container mt4">{renderWorkflowCards}</div>
          </>
        ),
      },
    ],
    [renderWorkflowCards, dispatch]
  );

  const handleSearchEvent = useCallback(
    (value: string) => {
      handleSearch(dispatch, query?.get('workflow_type'), value);
    },
    [dispatch, query]
  );

  return (
    <div
      className="__DashboardPageContainer__"
      data-testid="backoffice-dashboard-page"
    >
      <Breadcrumbs title1="Dashboard" href1="" dashboardPage />
      <div className="inner-container mt4">
        <h2 className="f2 center">Search Backoffice</h2>
        <div className="search-container">
          <Search
            enterButton
            placeholder="Search Backoffice"
            onSearch={value => dispatch(searchQueryUpdate(BACKOFFICE_SEARCH_NS, { q: value || undefined }))}
          />
        </div>
        <h2 className="f2 center mb4">Overview</h2>
        <EmptyOrChildren data={facets} title="0 Results">
          <LoadingOrChildren loading={loading}>
            <Tabs centered items={tabItems} />
          </LoadingOrChildren>
        </EmptyOrChildren>
      </div>
    </div>
  );
};

const mapStateToProps = (state: RootStateOrAny) => ({
  facets: state.backoffice.get('facets'),
  loading: state.backoffice.get('loading'),
  query: state.backoffice.get('query'),
});

export default connect(mapStateToProps)(DashboardPageContainer);
