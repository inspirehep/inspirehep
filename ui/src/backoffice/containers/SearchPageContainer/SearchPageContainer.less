.__SearchPageContainer__ {
  width: 100%;
  max-width: 2300px;

  .facet-category {
    font-size: 1rem;
    font-weight: 600;
  }

  .ant-tag {
    margin-right: 0;
  }
}

.waiting,
.running {
  color: #7b898a;
}
.error {
  color: #e74c3c;
}
.approval {
  color: #cfa90e;
}
.completed {
  color: #16a085;
}
