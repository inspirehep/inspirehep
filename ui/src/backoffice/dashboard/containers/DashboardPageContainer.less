.__DashboardPageContainer__ {
  width: 100%;

  .inner-container {
    display: flex;
    flex-direction: column;
    justify-content: center;

    .search-container {
      margin: 50px auto;
      padding: 15px;
      max-width: 1300px;
      width: 100%;
      display: flex;

      .ant-input {
        height: 3rem;
        border-top-right-radius: 0;
        border-bottom-right-radius: 0;
      }

      .ant-btn {
        width: 3rem;
        height: 3rem;
        border-top-left-radius: 0;
        border-bottom-left-radius: 0;
      }
    }

    .cards-container {
      margin: 20px auto 50px auto;
      display: flex;
      justify-content: center;
      flex-wrap: wrap;
      max-width: 1300px;

      .ant-card {
        margin: 20px 10px;
        width: 300px;
        border-width: 1px;
        transition: ease 0.25s;

        .ant-card-head {
          .ant-card-head-title p {
            text-transform: uppercase;
            text-wrap: wrap;
          }
        }

        .ant-card-body {
          border-color: white;
        }
      }
    }
  }
}

.ant-select-selection-item::first-letter,
.ant-select-item-option-content::first-letter {
  text-transform: capitalize;
}

// Color palette
.waiting,
.running {
  color: #7b898a;
}
.error,
.blocked {
  color: #e74c3c;
}
.halted,
.approval,
.matching {
  color: #cfa90e;
}
.completed {
  color: #16a085;
}
.initial {
  color: black;
}
.blue {
  color: #2980b9;
}
.processing {
  color: #0050b3;
}

.bg-error {
  background-color: #e74c3c;
}
.bg-completed {
  background-color: #16a085;
}
.bg-running {
  background-color: #b8b8b8;
}
.bg-halted,
.bg-approval {
  background-color: #f39c12;
}
.bg-waiting {
  background-color: #f5f5f5;
  font-family: Menlo, Monaco, Consolas, 'Courier New', monospace;
}
.bg-running {
  background-color: #c2c5c7;
}
.bg-processing {
  background-color: #b8b8b8;
}

.font-white {
  color: white !important;
}

.__DashboardPageContainer__ {
  .new-authors {
    &:hover {
      box-shadow: 0px 0px 10px 0px #3398db;
    }
    .ant-card-head {
      color: #3398db;
    }
  }

  .new-publisher-harvests {
    &:hover {
      box-shadow: 0px 0px 10px 0px #1abc9c;
    }
    .ant-card-head {
      color: #1abc9c;
    }
  }

  .publisher-updates {
    &:hover {
      box-shadow: 0px 0px 10px 0px #16a085;
    }
    .ant-card-head {
      color: #16a085;
    }
  }

  .author-updates {
    &:hover {
      box-shadow: 0px 0px 10px 0px #2980b9;
    }
    .ant-card-head {
      color: #2980b9;
    }
  }

  .new-arxiv-harvests {
    &:hover {
      box-shadow: 0px 0px 10px 0px #9c27b0;
    }
    .ant-card-head {
      color: #9c27b0;
    }
  }

  .new-literature-submissions {
    &:hover {
      box-shadow: 0px 0px 10px 0px #e67e22;
    }
    .ant-card-head {
      color: #e67e22;
    }
  }

  .arxiv-updates {
    &:hover {
      box-shadow: 0px 0px 10px 0px #8e44ad;
    }
    .ant-card-head {
      color: #8e44ad;
    }
  }

  .all-harvests {
    &:hover {
      box-shadow: 0px 0px 10px 0px #829091;
    }
    .ant-card-head {
      color: #829091;
    }
  }

  .merging {
    &:hover {
      box-shadow: 0px 0px 10px 0px #42b1b9;
    }
    .ant-card-head {
      color: #42b1b9;
    }
  }

  .matching {
    .ant-card-head {
      color: #5d35a7;
    }
  }

  .core-selection {
    &:hover {
      box-shadow: 0px 0px 10px 0px #ed9b16;
    }
    .ant-card-head {
      color: #ed9b16;
    }
  }

  .core-selection-approval {
    &:hover {
      box-shadow: 0px 0px 10px 0px #d3358c;
    }
    .ant-card-head {
      color: #d3358c;
    }
  }
}

.ant-tabs {
  max-width: 1300px;
  margin: 0 auto;
  .ant-tabs-tab,
  .ant-tabs-tab:focus-visible,
  .ant-tabs-tab:active {
    background: transparent !important;
    border-bottom: none !important;
    h3 {
      margin-bottom: 0;
      font-size: 1rem;
      font-weight: 400;
      color: black;
    }
  }
  .ant-tabs-tab.ant-tabs-tab-active,
  .ant-tabs-tab:focus-visible.ant-tabs-tab-active,
  .ant-tabs-tab:active.ant-tabs-tab-active {
    h3 {
      color: #1890ff;
    }
  }
}
