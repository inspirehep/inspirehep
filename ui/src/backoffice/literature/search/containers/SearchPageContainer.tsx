import React from 'react';
import { Row, Col, Select, Card } from 'antd';
import { connect, RootStateOrAny } from 'react-redux';
import { Action, ActionCreator } from 'redux';

import PaginationContainer from '../../../../common/containers/PaginationContainer';
import ResultsContainer from '../../../../common/containers/ResultsContainer';
import NumberOfResultsContainer from '../../../../common/containers/NumberOfResultsContainer';
import LoadingOrChildren from '../../../../common/components/LoadingOrChildren';
import DocumentHead from '../../../../common/components/DocumentHead';
import { SEARCH_PAGE_GUTTER } from '../../../../common/constants';
import { searchQueryUpdate } from '../../../../actions/search';
import { BACKOFFICE_LITERATURE_SEARCH_NS } from '../../../../search/constants';
import AggregationFiltersContainer from '../../../../common/containers/AggregationFiltersContainer';
import ResponsiveView from '../../../../common/components/ResponsiveView';
import DrawerHandle from '../../../../common/components/DrawerHandle';
import AggregationBox from '../../../../common/components/AggregationBox';
import ResultItem from '../../../common/components/ResultItem/ResultItem';
import Breadcrumbs from '../../../common/components/Breadcrumbs/Breadcrumbs';
import EmptyOrChildren from '../../../../common/components/EmptyOrChildren';

type BackofficeSearchPageProps = {
  loading: boolean;
  query: any;
  loadingAggregations: boolean;
  results: Map<string, any>;
  onSortByChange: (namespace: string, value: string) => void;
};

const META_DESCRIPTION = 'Find literature workflows in backoffice';
const TITLE = 'Search literature - Backoffice';

function renderWorkflowItem(item: Map<string, any>) {
  return <ResultItem item={item} key={item.get('id')} />;
}

const LiteratureSearchPageContainer = ({
  loading,
  query,
  loadingAggregations,
  results,
  onSortByChange,
}: BackofficeSearchPageProps) => {
  const renderAggregations = () => (
    <LoadingOrChildren loading={loadingAggregations}>
      <AggregationFiltersContainer
        namespace={BACKOFFICE_LITERATURE_SEARCH_NS}
      />
    </LoadingOrChildren>
  );
  return (
    <div
      className="__SearchPageContainer__"
      data-testid="backoffice-search-page"
    >
      <DocumentHead title={TITLE} description={META_DESCRIPTION} />
      <Breadcrumbs
        namespace={BACKOFFICE_LITERATURE_SEARCH_NS}
        title1="Search literature"
        href1="literature/search"
      />
      <Row justify="center">
        <EmptyOrChildren data={results} title="0 Results">
          <Col xs={24} lg={24} xl={24} xxl={24}>
            <Row className="mt3" gutter={SEARCH_PAGE_GUTTER} justify="start">
              <Col xs={0} lg={6}>
                <Card size="small">
                  <div style={{ padding: '0.5rem' }}>
                    <AggregationBox name="Sort by">
                      <Select
                        defaultValue="-_updated_at"
                        value={query?.get('ordering')}
                        style={{ width: '100%' }}
                        data-testid="select-sort-by"
                        options={[
                          { value: '-_updated_at', label: 'Most recent' },
                          { value: '_updated_at', label: 'Least recent' },
                        ]}
                        onChange={(value: string) =>
                          onSortByChange(BACKOFFICE_LITERATURE_SEARCH_NS, value)
                        }
                      />
                    </AggregationBox>
                  </div>
                  {renderAggregations()}
                </Card>
              </Col>
              <Col xs={24} lg={18}>
                <Row justify="space-between" wrap={false}>
                  <span className="mr2" />
                  <Col style={{ width: '55%' }}>
                    <NumberOfResultsContainer
                      namespace={BACKOFFICE_LITERATURE_SEARCH_NS}
                    />
                  </Col>
                  <Col
                    style={{
                      width: '29%',
                      paddingLeft: '5px',
                      fontWeight: 600,
                    }}
                  >
                    Action & Status
                  </Col>
                  <Col
                    style={{
                      width: '18%',
                      paddingLeft: '5px',
                      fontWeight: 600,
                    }}
                  >
                    Submission Info
                  </Col>
                  <Col
                    style={{
                      width: '18%',
                      paddingLeft: '5px',
                      fontWeight: 600,
                    }}
                  >
                    Subject Areas
                  </Col>
                </Row>
                <LoadingOrChildren loading={loading}>
                  <Row>
                    <Col xs={12} lg={0}>
                      <ResponsiveView
                        max="md"
                        render={() => (
                          <DrawerHandle
                            handleText="Filter"
                            drawerTitle="Filter"
                          >
                            <div style={{ padding: '0.5rem' }}>
                              <AggregationBox name="Sort by">
                                <Select
                                  defaultValue="-_updated_at"
                                  value={query?.get('ordering')}
                                  style={{ width: '100%' }}
                                  data-testid="select-sort-by"
                                  options={[
                                    {
                                      value: '-_updated_at',
                                      label: 'Most recent',
                                    },
                                    {
                                      value: '_updated_at',
                                      label: 'Least recent',
                                    },
                                  ]}
                                  onChange={(value: string) =>
                                    onSortByChange(
                                      BACKOFFICE_LITERATURE_SEARCH_NS,
                                      value
                                    )
                                  }
                                />
                              </AggregationBox>
                            </div>
                            {renderAggregations()}
                          </DrawerHandle>
                        )}
                      />
                    </Col>
                  </Row>
                  <Row>
                    <Col span={24}>
                      <ResultsContainer
                        namespace={BACKOFFICE_LITERATURE_SEARCH_NS}
                        renderItem={renderWorkflowItem}
                      />
                      <PaginationContainer
                        namespace={BACKOFFICE_LITERATURE_SEARCH_NS}
                      />
                    </Col>
                  </Row>
                </LoadingOrChildren>
              </Col>
            </Row>
          </Col>
        </EmptyOrChildren>
      </Row>
    </div>
  );
};

const stateToProps = (state: RootStateOrAny) => ({
  loading: state.search.getIn([
    'namespaces',
    BACKOFFICE_LITERATURE_SEARCH_NS,
    'loading',
  ]),
  loadingAggregations: state.search.getIn([
    'namespaces',
    BACKOFFICE_LITERATURE_SEARCH_NS,
    'loadingAggregations',
  ]),
  query: state.search.getIn([
    'namespaces',
    BACKOFFICE_LITERATURE_SEARCH_NS,
    'query',
  ]),
  results: state.search.getIn([
    'namespaces',
    BACKOFFICE_LITERATURE_SEARCH_NS,
    'results',
  ]),
});

export const dispatchToProps = (dispatch: ActionCreator<Action>) => ({
  onSortByChange(namespace: string, value: string) {
    dispatch(searchQueryUpdate(namespace, { ordering: value }));
  },
});

export default connect(
  stateToProps,
  dispatchToProps
)(LiteratureSearchPageContainer);
