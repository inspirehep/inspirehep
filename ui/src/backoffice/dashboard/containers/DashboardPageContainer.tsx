import React, { useEffect, useMemo, useState } from 'react';
import { Input, Tabs, Select } from 'antd';
import { Action, ActionCreator } from 'redux';
import { connect, RootStateOrAny } from 'react-redux';
import { List, Map } from 'immutable';

import './DashboardPageContainer.less';
import { isUserLoggedInToBackoffice } from '../../../actions/backoffice';
import EmptyOrChildren from '../../../common/components/EmptyOrChildren';
import LoadingOrChildren from '../../../common/components/LoadingOrChildren';
import { handleSearch } from '../../utils/utils';
import Breadcrumbs from '../../common/components/Breadcrumbs/Breadcrumbs';
import DocumentHead from '../../../common/components/DocumentHead';
import {
  BACKOFFICE_AUTHORS_SEARCH_NS,
  BACKOFFICE_LITERATURE_SEARCH_NS,
} from '../../../search/constants';
import { getConfigFor } from '../../../common/config';
import WorkflowCard from '../components/WorkflowCard';

interface DashboardPageContainerProps {
  dispatch: ActionCreator<Action>;
  authors: Map<string, any>;
  literature: Map<string, any>;
  loading: boolean;
}

const META_DESCRIPTION =
  'Welcome to the Backoffice Dashboard. Manage workflows, track statuses, and access important tools and resources to streamline your backoffice operations.';
const TITLE = 'Home - Backoffice';

const { Search } = Input;
const { Option } = Select;

const DashboardPageContainer = ({
  dispatch,
  authors,
  literature,
  loading,
}: DashboardPageContainerProps) => {
  const [searchNamespace, setSearchNamespace] = useState<
    typeof BACKOFFICE_AUTHORS_SEARCH_NS | typeof BACKOFFICE_LITERATURE_SEARCH_NS
  >(BACKOFFICE_AUTHORS_SEARCH_NS);

  useEffect(() => {
    dispatch(isUserLoggedInToBackoffice());
  }, [dispatch]);

  const workflowTypes: List<Map<string, any>> = useMemo(() => {
    const workflowTypePath = [
      '_filter_workflow_type',
      'workflow_type',
      'buckets',
    ];

    const getWorkflowTypeBuckets = (
      data: Map<string, any> | undefined
    ): List<Map<string, any>> => {
      return (
        (data?.getIn(workflowTypePath) as List<Map<string, any>>) || List()
      );
    };

    const authorsWorkflowTypes = getWorkflowTypeBuckets(authors);
    const literatureWorkflowTypes = getWorkflowTypeBuckets(literature);

    return authorsWorkflowTypes.concat(literatureWorkflowTypes);
  }, [authors, literature]);

  const renderWorkflowCards = useMemo(
    () =>
      workflowTypes?.map((type: Map<string, any>) => (
        <WorkflowCard
          key={type?.get('key')}
          type={type}
          statuses={
            type?.getIn(['status', 'buckets']) as List<Map<string, any>>
          }
        />
      )),
    [workflowTypes]
  );

  const tabItems = [
    {
      label: <h3>Collections</h3>,
      key: '1',
      children: (
        <div className="cards-container mt4">{renderWorkflowCards}</div>
      ),
    },
  ];

  return (
    <div
      className="__DashboardPageContainer__"
      data-testid="backoffice-dashboard-page"
    >
      <DocumentHead title={TITLE} description={META_DESCRIPTION} />
      <Breadcrumbs namespace="" title1="Dashboard" href1="" dashboardPage />
      <div className="inner-container mt4">
        <h2 className="f2 center">Search Backoffice</h2>
        <div className="search-container">
          <Search
            enterButton
            addonBefore={
              <Select value={searchNamespace} onChange={setSearchNamespace}>
                {getConfigFor('BACKOFFICE_LITERATURE_FEATURE_FLAG') && (
                  <Option value={BACKOFFICE_LITERATURE_SEARCH_NS}>
                    Literature
                  </Option>
                )}
                <Option value={BACKOFFICE_AUTHORS_SEARCH_NS}>Authors</Option>
              </Select>
            }
            placeholder="Search Backoffice"
            onSearch={(value) => handleSearch(dispatch, value, searchNamespace)}
          />
        </div>
        <h2 className="f2 center mb4">Overview</h2>
        <LoadingOrChildren loading={loading}>
          <EmptyOrChildren data={authors || literature} title="0 Results">
            <Tabs centered items={tabItems} />
          </EmptyOrChildren>
        </LoadingOrChildren>
      </div>
    </div>
  );
};

const mapStateToProps = (state: RootStateOrAny) => ({
  authors: state.backoffice.getIn(['dashboard', 'facets', 'authors']),
  literature: state.backoffice.getIn(['dashboard', 'facets', 'literature']),
  loading: state.backoffice.getIn(['dashboard', 'loading']),
});

export default connect(mapStateToProps)(DashboardPageContainer);
