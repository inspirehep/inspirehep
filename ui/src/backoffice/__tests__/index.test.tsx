import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter, Route } from 'react-router-dom';
import { fromJS } from 'immutable';
import { render } from '@testing-library/react';

import { getStore, getStoreWithState } from '../../fixtures/store';
import Backoffice from '..';
import DashboardPageContainer from '../containers/DashboardPageContainer/DashboardPageContainer';
import SearchPageContainer from '../containers/SearchPageContainer/SearchPageContainer';
import DetailPageContainer from '../containers/DetailPageContainer/DetailPageContainer';
import { BACKOFFICE_SEARCH, BACKOFFICE } from '../../common/routes';

describe('Backoffice', () => {
  const store = getStoreWithState({
    user: fromJS({
      loggedIn: true,
      data: {
        roles: ['cataloger'],
      },
    }),
    backoffice: fromJS({
      loggedIn: true,
      facets: fromJS({
        _filter_workflow_type: fromJS({
          workflow_type: fromJS({
            buckets: fromJS([
              {
                key: 'AUTHOR_CREATE',
                doc_count: 1,
                status: [{ key: 'error' }],
              },
              {
                key: 'AUTHOR_UPDATE',
                doc_count: 1,
              },
              {
                key: 'HEP_CREATE',
                doc_count: 1,
              },
            ]),
          }),
        }),
      }),
    }),
  });

  it('renders initial state', () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter initialEntries={[BACKOFFICE]}>
          <Backoffice />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('navigates to DashboardPageContainer when /backoffice/dashboard', () => {
    const { getByTestId } = render(
      <Provider store={getStore()}>
        <MemoryRouter initialEntries={[BACKOFFICE]}>
          <Route path={BACKOFFICE} component={DashboardPageContainer} />
        </MemoryRouter>
      </Provider>
    );

    expect(getByTestId('backoffice-dashboard-page')).toBeInTheDocument();
  });

  it('navigates to DetailPageContainer when /backoffice/:id', () => {
    const { getByTestId } = render(
      <Provider store={getStore()}>
        <MemoryRouter initialEntries={[`${BACKOFFICE}/:id`]}>
          <Route path={`${BACKOFFICE}/:id`} component={DetailPageContainer} />
        </MemoryRouter>
      </Provider>
    );

    expect(getByTestId('backoffice-detail-page')).toBeInTheDocument();
  });

  it('navigates to SearchPageContainer when /backoffice/search', () => {
    const { getByTestId } = render(
      <Provider store={getStore()}>
        <MemoryRouter initialEntries={[BACKOFFICE_SEARCH]}>
          <Route path={BACKOFFICE_SEARCH} component={SearchPageContainer} />
        </MemoryRouter>
      </Provider>
    );

    expect(getByTestId('backoffice-search-page')).toBeInTheDocument();
  });
});
