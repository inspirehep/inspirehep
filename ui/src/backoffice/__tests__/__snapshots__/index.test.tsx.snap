// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backoffice renders initial state 1`] = `
<div>
  <div
    class="w-100"
    data-testid="backoffice"
  >
    <div
      class="__DashboardPageContainer__"
      data-testid="backoffice-dashboard-page"
    >
      <div
        class="flex items-center justify-between"
      >
        <nav
          class="ant-breadcrumb mv4"
        >
          <ol>
            <li>
              <span
                class="ant-breadcrumb-link"
              >
                <a
                  href="/"
                >
                  <span
                    aria-label="home"
                    class="anticon anticon-home mr2"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="home"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                      />
                    </svg>
                  </span>
                   Inspirehep
                </a>
              </span>
              <span
                class="ant-breadcrumb-separator"
              >
                |
              </span>
            </li>
            <li>
              <span
                class="ant-breadcrumb-link"
              >
                <a
                  href="/backoffice"
                >
                  Backoffice
                </a>
              </span>
              <span
                class="ant-breadcrumb-separator"
              >
                |
              </span>
            </li>
            <li>
              <span
                class="ant-breadcrumb-link"
              >
                <a
                  href="/backoffice/"
                >
                  Dashboard
                </a>
              </span>
              <span
                class="ant-breadcrumb-separator"
              >
                |
              </span>
            </li>
          </ol>
        </nav>
      </div>
      <div
        class="inner-container mt4"
      >
        <h2
          class="f2 center"
        >
          Search Backoffice
        </h2>
        <div
          class="search-container"
        >
          <span
            class="ant-input-group-wrapper ant-input-search ant-input-search-with-button"
          >
            <span
              class="ant-input-wrapper ant-input-group"
            >
              <input
                class="ant-input"
                placeholder="Search Backoffice"
                type="text"
                value=""
              />
              <span
                class="ant-input-group-addon"
              >
                <button
                  class="ant-btn ant-btn-primary ant-input-search-button"
                  type="button"
                >
                  <span
                    aria-label="search"
                    class="anticon anticon-search"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="search"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </span>
        </div>
        <h2
          class="f2 center mb4"
        >
          Overview
        </h2>
        <div
          class="ant-tabs ant-tabs-top ant-tabs-centered"
        >
          <div
            class="ant-tabs-nav"
            role="tablist"
          >
            <div
              class="ant-tabs-nav-wrap"
            >
              <div
                class="ant-tabs-nav-list"
                style="transform: translate(0px, 0px);"
              >
                <div
                  class="ant-tabs-tab ant-tabs-tab-active"
                  data-node-key="1"
                >
                  <div
                    aria-controls="rc-tabs-test-panel-1"
                    aria-selected="true"
                    class="ant-tabs-tab-btn"
                    id="rc-tabs-test-tab-1"
                    role="tab"
                    tabindex="0"
                  >
                    <h3>
                      Collections
                    </h3>
                  </div>
                </div>
                <div
                  class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                />
              </div>
            </div>
            <div
              class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
            >
              <button
                aria-controls="rc-tabs-test-more-popup"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-hidden="true"
                class="ant-tabs-nav-more"
                id="rc-tabs-test-more"
                style="visibility: hidden; order: 1;"
                tabindex="-1"
                type="button"
              >
                <span
                  aria-label="ellipsis"
                  class="anticon anticon-ellipsis"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="ellipsis"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
          <div
            class="ant-tabs-content-holder"
          >
            <div
              class="ant-tabs-content ant-tabs-content-top"
            >
              <div
                aria-hidden="false"
                aria-labelledby="rc-tabs-test-tab-1"
                class="ant-tabs-tabpane ant-tabs-tabpane-active"
                id="rc-tabs-test-panel-1"
                role="tabpanel"
                tabindex="0"
              >
                <a
                  class="db w-100 tc f5 mt4"
                  data-testid="view-all"
                  href="/backoffice/search"
                >
                  View all
                </a>
                <div
                  class="cards-container mt4"
                >
                  <div
                    class="ant-card ant-card-bordered new-authors"
                  >
                    <div
                      class="ant-card-head"
                      style="text-align: center;"
                    >
                      <div
                        class="ant-card-head-wrapper"
                      >
                        <div
                          class="ant-card-head-title"
                        >
                          <div>
                            <p>
                              new authors
                            </p>
                            <p
                              class="f2 mb0 black"
                            >
                              1
                            </p>
                            <a
                              class="normal f6"
                              href="/backoffice/search?workflow_type=AUTHOR_CREATE"
                            >
                              View all
                            </a>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-card-body"
                    />
                  </div>
                  <div
                    class="ant-card ant-card-bordered author-updates"
                  >
                    <div
                      class="ant-card-head"
                      style="text-align: center;"
                    >
                      <div
                        class="ant-card-head-wrapper"
                      >
                        <div
                          class="ant-card-head-title"
                        >
                          <div>
                            <p>
                              author updates
                            </p>
                            <p
                              class="f2 mb0 black"
                            >
                              1
                            </p>
                            <a
                              class="normal f6"
                              href="/backoffice/search?workflow_type=AUTHOR_UPDATE"
                            >
                              View all
                            </a>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-card-body"
                    />
                  </div>
                  <div
                    class="ant-card ant-card-bordered new-literature-submissions"
                  >
                    <div
                      class="ant-card-head"
                      style="text-align: center;"
                    >
                      <div
                        class="ant-card-head-wrapper"
                      >
                        <div
                          class="ant-card-head-title"
                        >
                          <div>
                            <p>
                              new literature submissions
                            </p>
                            <p
                              class="f2 mb0 black"
                            >
                              1
                            </p>
                            <a
                              class="normal f6"
                              href="/backoffice/search?workflow_type=HEP_CREATE"
                            >
                              View all
                            </a>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="ant-card-body"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
